eth/db/header.py

```diff
@@ -83,11 +83,15 @@ def get_canonical_head(self) -> BlockHeaderAPI:
 
     @classmethod
     def _get_canonical_head(cls, db: DatabaseAPI) -> BlockHeaderAPI:
+        canonical_head_hash = cls._get_canonical_head_hash(db)
+        return cls._get_block_header_by_hash(db, Hash32(canonical_head_hash))
+
+    @classmethod
+    def _get_canonical_head_hash(cls, db: DatabaseAPI) -> Hash32:
         try:
-            canonical_head_hash = db[SchemaV1.make_canonical_head_hash_lookup_key()]
+            return cast(Hash32, db[SchemaV1.make_canonical_head_hash_lookup_key()])
```

Comment:
Christoph was recommending doing this instead. I guess type-checking is supposed to notice if you send a non-bytes value:

```suggestion
            return Hash32(db[SchemaV1.make_canonical_head_hash_lookup_key()])
```

YMMV, do what you like :)